python-frontend/src/main/java/org/sonar/python/semantic/v2/ProjectLevelTypeTable.java

```diff
@@ -67,10 +67,11 @@ public PythonType getType(List<String> typeFqnParts) {
       if (parent instanceof ModuleType moduleType) {
         TypeWrapper typeWrapper = moduleType.members().get(part);
         if (typeWrapper instanceof LazyTypeWrapper lazyTypeWrapper && !lazyTypeWrapper.isResolved()) {
-          if (i == typeFqnParts.size() - 1) {
+          if (i == typeFqnParts.size() - 1 && !(lazyTypeWrapper.hasImportPath(String.join(".", typeFqnParts)))) {
```

Comment:
Acknowledging the confusion when looked at this together, I'm thinking maybe we should extract this to another method, or at least add a tiny bit more info on the purpose of this in a comment (maybe updating the one on line 71).